### PR TITLE
En_Zob

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Zob/z_en_zob.c [hunk 1]
```diff
@@ -46,79 +49,725 @@ const ActorInit En_Zob_InitVars = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zob/func_80BA0C14.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zob/func_80BA0CF4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zob/EnZob_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zob/func_80BA0F64.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zob/func_80BA0FAC.s")
+static AnimationHeader* D_80BA10FC[9] = {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Zob/z_en_zob.c:72]:
```suggestion
static AnimationHeader* D_80BA10FC[] = {
```

**Review by engineer124 — APPROVED** [review]
Just one minor suggestion

File: src/overlays/actors/ovl_En_Zob/z_en_zob.c [hunk 2]
```diff
@@ -46,79 +49,725 @@ const ActorInit En_Zob_InitVars = {
...
+            this->unk_312 = -1;
+            func_80B9F7E4(this, 0, 2);
+            this->unk_304 = 5;
+            break;
+
+        default:
+            if (gSaveContext.weekEventReg[55] & 0x80) {
+                Actor_MarkForDeath(&this->actor);
+            }
+            this->actor.flags |= 0x10;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Zob/z_en_zob.c:139]:
```suggestion
            this->actor.flags |= ACTOR_FLAG_10;
```

File: src/overlays/actors/ovl_En_Zob/z_en_zob.c [hunk 3]
```diff
@@ -46,79 +49,725 @@ const ActorInit En_Zob_InitVars = {
...
+                }
+            }
+            break;
+    }
+}
+
+void func_80BA0610(EnZob* this, GlobalContext* globalCtx) {
+    func_80B9F86C(this);
+    if (Actor_ProcessTalkRequest(&this->actor, &globalCtx->state)) {
+        this->actor.flags &= ~0x10000;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Zob/z_en_zob.c:537]:
```suggestion
        this->actor.flags &= ~ACTOR_FLAG_10000;
```

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**